modules/update-using-custom-machine-config-pools-inheritance.adoc

```diff
@@ -0,0 +1,208 @@
+// Module included in the following assemblies:
+//
+// * updating/updating_a_cluster/update-using-custom-machine-config-pools.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="update-using-custom-machine-config-pools-mcp-inheritance_{context}"]
+= Managing machine configuration inheritance for a worker pool canary
+
+You can configure a machine config pool (MCP) canary to inherit any `MachineConfig` assigned to an existing MCP. 
+This configuration is useful when you want to use an MCP canary to test as you update the nodes of an existing MCP, one at a time.
+
+
+.Prerequisites
+
+* You have created one or more custom machine config pools (MCP).
+
+.Procedure
+
+. Create a secondary MCP as described in the following two steps:
+.. Save the following configuration file as `machineConfigPool.yaml`.
++
+.Example `machineConfigPool` YAML
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: worker-perf
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {
+         key: machineconfiguration.openshift.io/role,
+         operator: In,
+         values: [worker,worker-perf]
+        }
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker-perf: "" 
+----
++
+.. Create the new machine config pool by running the following command:
++
+[source,terminal]
+----
+$ oc create -f machineConfigPool.yaml
+----
++
+.Example output
++
+[source,terminal]
+----
+machineconfigpool.machineconfiguration.openshift.io/worker-perf created
+----
+
+. Add some machines to the secondary MCP.
+The following example labels the worker nodes, `worker-b` and `worker-c` to the MCP `worker-perf`:
++
+[source,terminal]
+----
+$ oc label node worker-b node-role.kubernetes.io/worker-perf=''
+$ oc label node worker-c node-role.kubernetes.io/worker-perf=''
+----
+
+. Create a new `MachineConfig` for the MCP `worker-perf` as described in the following two steps:
+.. Save the following `MachineConfig` example as `new-machineconfig.yaml`.
++
+.Example `MachineConfig` YAML
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: workers-perf
+  name: 06-kdump-enable-workers-perf
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - enabled: true
+        name: kdump.service
+  kernelArguments:
+    - crashkernel=512M
+----
+
+.. Apply the `MachineConfig` by running the following command:
++
+[source,terminal]
+----
+$ oc create -f new-machineconfig.yaml
+----
+
+. Create the new canary MCP and add machines from the MCP you created in the previous steps.
+The following example creates the an MCP called `worker-perf-canary` MCP, and adds machines from the `worker-perf` to it. 
+
+.. Label the canary worker node `worker-a` by running the following command: 
++
+[source,terminal]
+----
+$ oc label node worker-a node-role.kubernetes.io/worker-perf-canary=''
```

Comment:
It says `Create the new canary MCP and add machines from the MCP you created in the previous steps.`

I understand that we need to add to the new canary MCP the machines that we added to the "worker-perf" MCP. Is it like that?

We added worker-b and worker-c to the "worker-perf" MCP, but in this example we are adding "worker-a" to the new canary MCP. Is it ok? 

We should add worker-b or worker-c to the new canary instead of worker-a, right? (or add worker-a to the initial worker-perf MCP too at the beginning of the example)

We need to remove the worker-perf label after adding the worker-perf-canary label too, because a node cannot belong to 2 custom MCPs at the same time.

Reply:
Do we need to make it clear that :  all workers are member of workers-perf MCP firstly, then we move one of them to workers-perf-canary?

Reply:
Hello! No, it is not necessary to remove the "worker" label, just the "worker-pef" label.

Reply:
@sergiordlr Added a step to remove the `worker-perf` label